### ISSUE TITLE
Refactoring of schema API

### ISSUE THREAD

**Issue by lvca** [ArcadeData/arcadedb#1102]
### Discussed in https://github.com/ArcadeData/arcadedb/discussions/1101

<div type='discussions-op-text'>

<sup>Originally posted by **lvca** May 25, 2023</sup>
Right now we have many methods to create document, vertex and edge types. They originally were designed to allow different parameters, but it would be better to use builder instead. For example, this method:

```java
DocumentType createDocumentType(String typeName, List<Bucket> buckets, int pageSize);
```

Could be written with a Builder like this:

```java
buildDocumentType().withName(typeName).withBuckets(buckets).withPageSize(pageSize).create()
```

I'd like to refactor this API and mark the (many) methods with a different signature as `@Deprecated` and keep them for a while before removing them.

Any thoughts?
</div>

**Comment by lvca** [comment]
All the components and tests that use the now deprecated API must use the new builder.

**Comment by lvca** [comment]
Fixed in 9ab01fec

**Comment by lvca** [comment]
We need to do the same ith indexes. Reopening the issue.
